src/core/thread/mle.cpp

```diff
@@ -3071,6 +3070,48 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
         ExitNow(error = kErrorDrop);
     }
 
+    // In case key sequence is larger, we determine whether to adopt it
+    // or not. The `Handle{MleMsg}()` methods set the `rxInfo.mClass`
+    // based on the message command type and the included TLVs. If
+    // there is any error during parsing of the message the `mClass`
+    // remains as its default value of `RxInfo::kUnknown`. Message
+    // classes are determined based on this:
+    //
+    // Authoritative : Larger key seq MUST be adopted.
+    // Tentative     : Larger key seq MUST NOT be adopted.
+    // Peer          : If from a known neighbor
+    //                    If difference is 1, adopt
+    //                    Otherwise try to re-sync with neighbor.
+
+    if (keySequence > Get<KeyManager>().GetCurrentKeySequence())
```

Comment:
would it be more efficient storing the result of "Get<KeyManager>().GetCurrentKeySequence()" in a variable so this can be re-used in the second usage in line 3101? Avoids a double 'Get' in code. Just curious (don't want to violate any coding guidelines)

Reply:
It won't make any noticeable difference: No difference in code-size (may actually increase code-size in some cases if the local variable ends up on stack instead of in a register). Also no real noticeable impact on speed of execution. 

All these simple `Get<Type>` are fully inlined, same with `GetCurrentKeySequence()`. So the whole statement ends up becoming a simple instruction to access a member variable (access some address/memory).